clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -2403,17 +2404,40 @@ private ClusterResourceListeners configureClusterResourceListeners(Deserializer<
         return clusterResourceListeners;
     }
 
-    private void close(long timeoutMs, boolean swallowException) {
+    private void close(Duration timeout, boolean swallowException) {
         log.trace("Closing the Kafka consumer");
         AtomicReference<Throwable> firstException = new AtomicReference<>();
-        try {
-            if (coordinator != null)
-                coordinator.close(time.timer(Math.min(timeoutMs, requestTimeoutMs)));
-        } catch (Throwable t) {
-            firstException.compareAndSet(null, t);
-            log.error("Failed to close coordinator", t);
+
+        final Timer closeTimer = (time == null) ? new SystemTime().timer(Math.min(timeout.toMillis(), requestTimeoutMs)) : time.timer(Math.min(timeout.toMillis(), requestTimeoutMs));
+        // Close objects with a timeout. The timeout is required because fetcher makes request to the server in the
+        // process of closing which may not respect the overall timeout defined for closing the consumer.
+        if (coordinator != null) {
+            try {
+                coordinator.close(closeTimer);
```

Comment:
FYI reviewer

Note that this is a blocking call. It sends a LEAVE_GROUP request and waits for it's completion before proceeding ahead.

Reply:
note: I think it also tries blocking commits the offset during the close.

Reply:
yes you are right @philipnee and I believe the code comment above referring "send requests to the server" covers that scenario. Do you want to suggest some action here?

Reply:
nope, just wanna to clarify. thanks.